### PR TITLE
[runtime env][java] Support runtime env config in Java

### PR DESCRIPTION
**Pull request by SongGuyang**
Signed-off-by: 久龙 <guyang.sgy@antfin.com>

## Why are these changes needed?

Support job level and task/actor level runtime env config eg. `setupTimeoutSeconds` and `eagerInstall`.

### CODE REVIEW

File: java/api/src/main/java/io/ray/api/runtimeenv/RuntimeEnv.java [hunk 1]
```diff
@@ -64,6 +72,13 @@ public interface RuntimeEnv {
    */
   public String serialize() throws RuntimeEnvException;
 
+  /**
+   * Whether the runtime env is empty.
+   *
+   * @return
+   */
+  boolean empty();
```
Comment (jovany-wang) [inline java/api/src/main/java/io/ray/api/runtimeenv/RuntimeEnv.java:80]: isEmpty() is a batter practice in Java.

File: java/api/src/main/java/io/ray/api/runtimeenv/RuntimeEnv.java [hunk 2]
```diff
@@ -83,6 +98,20 @@ public static RuntimeEnv deserialize(String serializedRuntimeEnv) throws Runtime
...
+   * Set runtime env config.
+   *
+   * @param runtimeEnvConfig
+   */
+  public void setConfig(RuntimeEnvConfig runtimeEnvConfig);
+
+  /**
+   * Get runtime env config.
+   *
+   * @return
```
Comment (jovany-wang) [inline java/api/src/main/java/io/ray/api/runtimeenv/RuntimeEnv.java:111]:
```suggestion
   * @return The runtime env config.
```
or remove this line

File: java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java [hunk 3]
```diff
@@ -37,7 +43,7 @@ public void setJsonStr(String name, String jsonStr) throws RuntimeEnvException {
     try {
       node = JsonLoader.fromString(jsonStr);
     } catch (IOException e) {
-      throw new RuntimeException(e);
+      throw new RuntimeEnvException("Setting json field error", e);
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java:46]:
```suggestion
      throw new RuntimeEnvException("Failed to set json field.", e);
```

Comment (jovany-wang) [reply java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java:46]: That's what we always write.

**Review by jovany-wang — APPROVED** [review]
LGTM

**Comment by SongGuyang** [comment]
The failed tests are not related.

**Pull request merged**